packages/nextjs/app/layout.tsx

```diff
@@ -0,0 +1,50 @@
+import "@rainbow-me/rainbowkit/styles.css";
+import { Metadata } from "next";
+import { AppComponent } from "~~/components/AppComponent";
+import "~~/styles/globals.css";
+
+const baseUrl = process.env.NEXT_PUBLIC_VERCEL_URL ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}/` : "/";
+const image = "thumbnail.jpg";
+const imageUrl = baseUrl + image;
+export const metadata: Metadata = {
+  title: {
+    default: "Scaffold-ETH 2 App",
+    template: "%s | Scaffold-ETH 2",
+  },
+  description: "Built with 🏗 Scaffold-ETH 2",
+  openGraph: {
+    title: {
+      default: "Scaffold-ETH 2 App",
+      template: "%s | Scaffold-ETH 2",
+    },
+    description: "Built with 🏗 Scaffold-ETH 2",
+    images: [
+      {
+        url: imageUrl,
+      },
+    ],
+  },
+  twitter: {
+    card: "summary_large_image",
+    images: [imageUrl],
+    title: {
+      default: "Scaffold-ETH 2",
+      template: "%s | Scaffold-ETH 2",
+    },
+    description: "Built with 🏗 Scaffold-ETH 2",
+  },
+  icons: {
+    icon: [{ url: "/favicon.png", sizes: "32x32", type: "image/png" }],
+  },
+};
```

Comment:
Add empty lines before meaningful blocks if possible. I thought prettier adds it 🤷‍♂️